pkgs/tools/text/gawk/gawk-with-extensions.nix

```diff
@@ -0,0 +1,12 @@
+{ runCommand, gawk, extensions ? [ ], makeWrapper }:
+
+runCommand "gawk-with-extensions" {
+  buildInputs = [ makeWrapper gawk ] ++ extensions;
+} ''
+  mkdir -p $out/bin
+  for i in ${gawk}/bin/*; do
+    name="$(basename "$i")"
+    makeWrapper $i $out/bin/$name \
+      --suffix AWKLIBPATH : "${gawk}/lib/gawk:''${AWKLIBPATH:-}"
```

Comment:
`--suffix` means that the wrapper will always override user setting for AWKLIBPATH?